.github/workflows/test-python3-support.yml

```diff
@@ -0,0 +1,135 @@
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          wget -qO - http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/xUbuntu_18.04/Release.key | sudo apt-key add -
+          echo 'deb http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/xUbuntu_18.04 ./' | sudo tee /etc/apt/sources.list.d/syslog-ng-obs.list
+
+          sudo apt-get update -qq
+          sudo apt-get install -y --no-install-recommends \
+            autoconf-archive \
+            bison \
+            docbook-xsl \
+            flex \
+            gradle \
+            libcap-dev \
+            libcurl4-openssl-dev \
+            libdbd-sqlite3 \
+            libdbi0-dev \
+            libesmtp-dev \
+            libgeoip-dev \
+            libglib2.0-dev \
+            libhiredis-dev \
+            libivykis-dev \
+            librabbitmq-dev \
+            libmongoc-dev \
+            libnet1-dev \
+            libriemann-client-dev \
+            librdkafka-dev \
+            libwrap0-dev \
+            pkg-config \
+            sqlite3 \
+            xsltproc \
+            criterion-dev \
+            libmaxminddb-dev \
+            libxml2-utils \
+            doxygen \
+            libsnmp-dev \
+            python3-pip \
+            python3-setuptools
+
+          python3 -m pip install --user -r requirements.txt
+
+          javac -version
+
+      - name: autogen
+        run:
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          ./autogen.sh
+
+      - name: configure
+        run: |
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          ./configure \
+            --with-python=3 \
+            --with-ivykis=system \
+            --disable-sun-streams \
+            --disable-systemd \
+            --disable-pacct \
+            --disable-smtp \
+            --disable-geoip \
+            --enable-all-modules \
+            |& tee -i configure.log
+
+      - name: "artifact: configure.log"
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: configure.log
+          path: configure.log
+
+      - name: build
+        run: |
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          make --keep-going -j $(nproc) || \
+            { \
```

Comment:
[optional] I would rather forget about this trial and error, instead we should fix missing build dependencies.